### PR TITLE
Add a Dockerfile for aggregation_job_creator

### PR DESCRIPTION
**Pull request by divergentdave**
This adds a Dockerfile that builds the aggregation_job_creator binary. Thanks to caching, our two Dockerfiles can reuse the lion's share of the work they do.

### CODE REVIEW

**Review by BranLwyd — APPROVED** [review]

File: .github/workflows/ci-build.yml [hunk 1]
```diff
@@ -43,3 +43,6 @@ jobs:
     - env:
         DOCKER_BUILDKIT: 1
       run: docker build .
```
Comment (tgeoghegan) [inline .github/workflows/ci-build.yml:45]: We should rename the existing Dockerfile to `Dockerfile.aggregator` and use `-f` here, for consistency and to disambiguate.

File: Dockerfile.aggregation_job_creator [hunk 2]
```diff
@@ -0,0 +1,14 @@
+FROM rust:1.60.0-alpine as builder
+RUN apk add libc-dev
+
+WORKDIR /src
+COPY Cargo.toml /src/Cargo.toml
+COPY Cargo.lock /src/Cargo.lock
+COPY janus_server /src/janus_server
+COPY monolithic_integration_test /src/monolithic_integration_test
+COPY test_util /src/test_util
+RUN --mount=type=cache,target=/usr/local/cargo/registry --mount=type=cache,target=/src/target cargo build --release --bin aggregation_job_creator && cp /src/target/release/aggregation_job_creator /aggregation_job_creator
```
Comment (tgeoghegan) [inline Dockerfile.aggregation_job_creator:10]: Is the only difference between the two dockerfiles `--bin aggregator` vs. `--bin aggregation_job_creator` and thenwhich path gets copied in from the builder image? If so, maybe we can use a single Dockerfile that takes a parameter for image name.

File: Dockerfile.aggregation_job_creator [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+FROM alpine:3.15.4
```
Comment (tgeoghegan) [inline Dockerfile.aggregation_job_creator:12]: The `aggregator` Dockerfile copes in `/src/db/schema.sql`. Do we not need that here?

**Review by tgeoghegan — CHANGES_REQUESTED** [review]

File: Dockerfile.aggregation_job_creator (hunk 2 above)
Comment (tgeoghegan) [reply Dockerfile.aggregation_job_creator:10]: Obviously this suggestion contradicts my other comment on `.github/workflows/ci-build.yml`. I'm comfortable with either a single or dual Dockerfile solution, so please move ahead with whichever one you prefer.

File: Dockerfile.aggregation_job_creator (hunk 3 above)
Comment (divergentdave) [reply Dockerfile.aggregation_job_creator]: It doesn't get used in the container at runtime, as we only copy it into the container for use by our Terraform external data provider. This is now moot as I'm going with a single Dockerfile.

File: Dockerfile.aggregation_job_creator (hunk 2 above)
Comment (divergentdave) [reply Dockerfile.aggregation_job_creator]: I went with one parameterized file, nice idea! Build arguments can't be directly used in the ENTRYPOINT, unfortunately, so I decided to copy the argument into an environment variable (which becomes part of the image) and use that in a shell command. The alternate solution would be to rename the binary to something constant when copying it, and then put that common name in the ENTRYPOINT. I decided against that, as I think one PID plus a few megabytes of memory is worth the clarity of having argv[0] be the "real" binary name.

**Review by tgeoghegan — APPROVED** [review]

**Pull request merged**